api_v3/lib/types/KalturaUserEntry.php

```diff
@@ -118,6 +118,16 @@ public function toInsertableObject ( $object_to_fill = null , $props_to_skip = a
 			}
 			$object_to_fill->setKuserId($kuser->getKuserId());
 		}
+
+		$c = new Criteria();
+		$c->add(UserEntryPeer::KUSER_ID, $object_to_fill->getKuserId());
+		$c->add(UserEntryPeer::ENTRY_ID, $this->entryId);
+		$userEntry = UserEntryPeer::doSelect($c);
+		if (count($userEntry) > 0)
```

Comment:
Won't it prevent more than one anonymous user-entry?
Also, it will prevent adding non-quiz user-entry objects, so please add type to the query.